pkg/storage/stores/shipper/index/table_manager.go

```diff
@@ -163,25 +168,31 @@ func (tm *TableManager) handoverIndexesToShipper(force bool) {
 	tm.tablesMtx.RLock()
 	defer tm.tablesMtx.RUnlock()
 
-	level.Info(util_log.Logger).Log("msg", "handing over indexes to shipper")
+	level.Info(tm.logger).Log("msg", "handing over indexes to shipper")
 
 	for _, table := range tm.tables {
 		err := table.HandoverIndexesToShipper(force)
 		if err != nil {
 			// continue handing over other tables while skipping cleanup for a failed one.
-			level.Error(util_log.Logger).Log("msg", "failed to handover index", "table", table.name, "err", err)
+			level.Error(tm.logger).Log("msg", "failed to handover index", "table", table.name, "err", err)
 			continue
 		}
 
 		err = table.Snapshot()
 		if err != nil {
 			// we do not want to stop handing over of index due to failures in snapshotting them so logging just the error here.
-			level.Error(util_log.Logger).Log("msg", "failed to snapshot table for reads", "table", table.name, "err", err)
+			level.Error(tm.logger).Log("msg", "failed to snapshot table for reads", "table", table.name, "err", err)
 		}
 	}
 }
 
 func (tm *TableManager) loadTables() (map[string]*Table, error) {
+	// as we are running multiple instances of TableManger - one for each period
+	// existing tables should be migrated to period specific directory
+	if err := migrateTables(tm.cfg.IndexDir, tm.tableRange, tm.logger); err != nil {
```

Comment:
I began with this approach since overlapping chunks were being indexed on both sides of a period boundary which would've caused conflicting writes from multiple table managers. I think we can remove this now given this problem is fixed with #8251